### DISCUSSION TITLE
Process function refactoring.

### DISCUSSION THREAD

**Discussion by st3rv04ka** [projectdiscovery/httpx#688] [Ideas]
I see that struct scanopts has clone() function, but it's very hard to maintain this func, because if we add some new arg, we need to change this function every time. 

My propose:

Remove clone() function and add URI string param to the process function.

I can help with it, but what do you think about it?

**Comment by Mzack9999** [comment]
@st3rv04ka Thanks for opening this discussion. Assuming we use a URL syntax to express values, wouldn't this require parsing and assigning values to respective fields and thus, need to be updated at each new addition/removal anyway?

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
